app/Http/Controllers/CartController.php

```diff
@@ -49,12 +47,10 @@ public function index(Request $request): View
 
     public function add(string $id, string $type, Request $request): RedirectResponse
     {
-        // Validar que el tipo sea válido
         if (! in_array($type, ['instrument', 'lesson'])) {
             return back()->withErrors('Invalid product type.');
         }
 
-        // Agregar el producto al carrito
         $cartItems = $request->session()->get('cart_items', []);
         $cartItems[] = ['id' => $id, 'type' => $type];
         $request->session()->put('cart_items', $cartItems);
```

Comment:
You are using this logic in several other places, create a method for this inside Controllers/Util aswell


```php
        $cartItems = $request->session()->get('cart_items', []);
        $cartItems[] = ['id' => $id, 'type' => $type];
        $request->session()->put('cart_items', $cartItems);
```